File: bin/Debug/net5.0/v4.4.2/lgcy-on-lgcy/examples/t_api-captcha-mvc5-api_basics/csharp/Controllers/ExampleController.cs
using System.Web.Mvc;

using BotDetect.Web.Mvc;

using Mvc50BasicCaptchaExampleCSharp.Models;


namespace Mvc50BasicCaptchaExampleCSharp.Controllers
{
    public class ExampleController : Controller
    {
        //
        // GET: /Example/

        public ActionResult Index()
        {
            return View();
        }


        //
        // POST: /Example/

        [HttpPost]
        [CaptchaValidationActionFilter("CaptchaCode", "ExampleCaptcha", "Incorrect!")]
        public ActionResult Index(ExampleModel model)
        {
            MvcCaptcha.ResetCaptcha("ExampleCaptcha");
            return View(model);
        }

    }
}

File: bin/Debug/net5.0/v4.4.2/lgcy-on-lgcy/examples/t_api-captcha-mvc5-api_basics/vbnet/Controllers/ExampleController.vb
Imports BotDetect.Web.Mvc
Imports System.Web.Mvc

Namespace Mvc50BasicCaptchaExampleVBNet
    Public Class ExampleController
        Inherits System.Web.Mvc.Controller

        '
        ' GET: /Example

        Function Index() As ActionResult
            Return View()
        End Function


        '
        ' POST: /Example/

        <HttpPost()> _
        <CaptchaValidationActionFilter("CaptchaCode", "ExampleCaptcha", "Incorrect!")> _
        Public Function Index(ByVal model As ExampleModel) As ActionResult
            MvcCaptcha.ResetCaptcha("ExampleCaptcha")
            Return View(model)
        End Function

    End Class
End Namespace